checker/check-new-request-non-path-default-parameter.go

```diff
@@ -13,26 +13,35 @@ func NewRequestNonPathDefaultParameterCheck(diffReport *diff.Diff, _ *diff.Opera
 		if pathItem.ParametersDiff == nil || pathItem.Revision == nil {
 			continue
 		}
-		for paramLoc, paramNameList := range pathItem.ParametersDiff.Added {
+
+		for paramLoc := range pathItem.ParametersDiff.Added {
 			if paramLoc == "path" {
 				continue
 			}
+
 			for _, param := range pathItem.Revision.Parameters {
-				if !paramNameList.Contains(param.Value.Name) {
-					continue
-				}
+
```

Comment:
The removal of this code makes the check incorrect.
The level will be `ERR` even if the parameter wasn't added.
I'm creating a fix.

Reply:
I was under the impression that it is redundant because we are iterating on pathItem.ParametersDiff.Added
Can you replicate the issue with a unit test?

Reply:
We do that, but since the `pathItem.Revision.Parameters` field is a slice and not a map, we traverse through it to find the `ParameterRef` object that matches the name we found in the `Added` map.
[PR](https://github.com/Tufin/oasdiff/pull/384).